### PR TITLE
Document local installation for development purposes in CONTRIBUTING.md

### LINKED ISSUE #324: Document local installation for development purposes in CONTRIBUTING.md
As for now I was using `python setup.py install`... but thanks to @BartoszSiwinski I realized that it could be done with `pip install -e .`. Let's research upsides and downsides of each and decide as to which one (maybe both?) to put in our guidelines.

### PR DESCRIPTION
**Pull request by amalota**

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 - fixed: `git machete` now correctly displays only available commands (without help topics)
 - fixed: `git machete traverse -y -M` no longer exits with error
 - fixed: `git machete` commands no longer exit with stack trace when interrupted by Ctrl+C
+- improved: instructions on how to install `git-machete` locally for development purposes
```
Comment (PawelLipski) [inline RELEASE_NOTES.md]: Not worth mentioning in release notes. In general, let's not include stuff that's not relevant for everyday users (excl. developers/contributors)

Comment (PawelLipski) [reply RELEASE_NOTES.md]: I sometimes put stuff like `fixed: Homebrew release process` in 3.12.4 but mostly coz nothing else happened in that release, just to put anything not to break the scripts xD

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -29,18 +29,28 @@ ln -s ../../ci/checks/run-all-checks.sh .git/hooks/pre-commit
 ```
 
 
-## Run tests locally
+## Local installation for development purposes
```
Comment (PawelLipski) [inline CONTRIBUTING.md]: :benissimo:

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -29,18 +29,28 @@ ln -s ../../ci/checks/run-all-checks.sh .git/hooks/pre-commit
...
 
-It's also possible to execute tests graphically in PyCharm (right-clicking on the test file
-or clicking the green triangle on the left of the test case name), which can be very useful when debugging a single test case.
-To do so, make sure that the libraries from [requirements.testenv.txt](requirements.testenv.txt) are installed in the current venv.
+
+### IDE (PyCharm)
+
+To execute tests in IDE (e.g. PyCharm -> right-clicking on the test file or clicking the green triangle on the left of the test case name),
+you must first have virtual environment. You can create one named `venv` by executing `tox -e venv` or use existing environment
+(need to install requirements from [requirements.testenv.txt](requirements.testenv.txt)). Lastly, aetup `python` interpreter for your project in the IDE, e.g. `.tox/venv/bin/python`.
```
Comment (PawelLipski) [inline CONTRIBUTING.md]:
```suggestion
(need to install requirements from [requirements.testenv.txt](requirements.testenv.txt)). Lastly, setup `python` interpreter for your project in the IDE, e.g. `.tox/venv/bin/python`.
```

**Comment by PawelLipski** [comment]
I'd also include the scenario where git-machete is installed for the entire user (and not just in the given venv), via plain `pip install --user -e .`. This has the upside that you can dogfood the tool the entire time. Just a single caveat that you need to add the site-package installation path to PATH, to be found via [site.USER_BASE](https://docs.python.org/3.11/library/site.html#site.USER_BASE) + `/bin`, e.g. `/Users/pawel_pawellipski/Library/Python/3.9/bin`

**Comment by amalota** [comment]
> I'd also include the scenario where git-machete is installed for the entire user (and not just in the given venv), via plain `pip install --user -e .`. This has the upside that you can dogfood the tool the entire time. Just a single caveat that you need to add the site-package installation path to PATH, to be found via [site.USER_BASE](https://docs.python.org/3.11/library/site.html#site.USER_BASE) + `/bin`, e.g. `/Users/pawel_pawellipski/Library/Python/3.9/bin`

@PawelLipski From what i read, using `--user` inside venv is not recommended. I also was not able to install it after adding site-package installatiion path to PATH, did you manage to get it working?

**Comment by PawelLipski** [comment]
> From what i read, using --user inside venv is not recommended

I mean, `pip install --user -e .` **outside** of a venv, just using system-wide `pip`/`python`

**Pull request merged**